lib/TBDGen/TBDGen.cpp

```diff
@@ -530,8 +530,10 @@ void TBDGenVisitor::addAutoDiffLinearMapFunction(AbstractFunctionDecl *original,
       config.parameterIndices,
       original->getInterfaceType()->castTo<AnyFunctionType>());
   Mangle::ASTMangler mangler;
-  AutoDiffConfig silConfig{loweredParamIndices, config.resultIndices,
-                           config.derivativeGenericSignature};
+  AutoDiffConfig silConfig{
```

Comment:
Can you create a new SILDeclRef-like abstraction that can allow this logic to be shared between TBDGen and SIL?

Reply:
Yeah, that's what I meant. TBDGen should not duplicate any logic from SILGen/IRGen, except for the top-level visiting of decls. Mangling and visibility should be computed by shared code, or else you'll hit frequent issues where the two are out of sync.

Reply:
Got it, thanks! I filed SR-13269 to track this issue so we can look into it later, if that's okay.